apps/apps_publisher.py

```diff
@@ -43,12 +44,19 @@ def tag(self, apps: ComposeApps, version: str):
 
     def __tag(self, app: ComposeApps.App, tag: str):
         changed = False
-        for _, service_cfg in app.services():
-            image_url = expandvars(service_cfg['image'])
+        for service_name, service_cfg in app.services():
+            image_url_template = service_cfg.get('image')
+            if not image_url_template:
+                raise Exception('Mandatory element `image:` is missing in the app service config; app: {}, service: {}'
+                                .format(app.name, service_name))
+            image_url = expandvars(image_url_template)
             logger.info('Service image url: %s', image_url)
-            if not image_url.startswith(self._image_base_url):
-                # TODO: verify if URL points to some other factory
+            if not image_url.startswith(self._registry_host):
+                # image url points to non-Foundries docker registry
                 continue
+            if not image_url.startswith(self._image_base_url):
+                raise Exception('Image url refers to an image that does not belong to the given Factory;'
+                                ' url: {}, factory: {}'.format(image_url, self._factory))
```

Comment:
I think this means you could do something like:
~~~
 image: hub.foundries.io/lmp/fiotest:postmerge
~~~

Reply:
or if a build is for msul-dev01 factory and I am referring to an image of msul-dev02 (we can tag only those images that we build).

Reply:
we need to allow customers to use the "lmp" factory containers.

Reply:
allow to use but not tag the image? so just skip/continue if the logic finds URL referring to the "lmp" factory container?